docs/fitting.rst

```diff
@@ -15,6 +15,177 @@ initial guesses.  This model is then actually fit to the spectrum's ``flux``,
+   >>> from specutils.manipulation import noise_region_uncertainty
+   >>> from specutils.fitting import find_lines_derivative
+
+   >>> # Define the spectrum
+   >>> np.random.seed(42)
+   >>> g1 = models.Gaussian1D(1, 4.6, 0.2)
+   >>> g2 = models.Gaussian1D(2.5, 5.5, 0.1)
+   >>> g3 = models.Gaussian1D(-1.7, 8.2, 0.1)
+   >>> x = np.linspace(0, 10, 200)
+   >>> y = g1(x) + g2(x) + g3(x) + np.random.normal(0., 0.2, x.shape)
+   >>> spectrum = Spectrum1D(flux=y*u.Jy, spectral_axis=x*u.um)
+
+   >>> # Define a noise region for adding the uncertainty
+   >>> noise_region = SpectralRegion(0*u.um, 3*u.um)
+
+   >>> # Derivative method
+   >>> lines = find_lines_derivative(spectrum, flux_threshold=0.75)
+
+   >>> emission_lines = lines[lines['line_type'] == 'emission'] #doctest:+SKIP
+         line_center    line_type line_center_index
+              um                                   
+      ----------------- --------- -----------------
+      4.522613065326634  emission                90
+      5.477386934673367  emission               109
+
+   >>> absorption_lines = lines[lines['line_type'] == 'absorption'] #doctest:+SKIP
+         line_center    line_type  line_center_index
+              um                                    
+      ----------------- ---------- -----------------
+      8.190954773869347 absorption               163
+
+
+Parameter Estimation
+--------------------
+
+Given a spectrum with a set of lines, the `~specutils.fitting.estimate_line_parameters`
+can be called to estimate the `~astropy.modeling.Model` parameters given a spectrum.
+
+For the `~astropy.modeling.functional_models.Gaussian1D`, 
+`~astropy.modeling.functional_models.Voigt1D`, and
+`~astropy.modeling.functional_models.Lorentz1D` models, there are predefined estimators for each 
+of the parameters. For all other models one must define the estimators (see example below).
+
+For example:
+
+.. code-block:: python
+
+   >>> import numpy as np
+   >>> from astropy.modeling import models
+   >>> import astropy.units as u
+   >>> from specutils import Spectrum1D
+   >>> from specutils.fitting import estimate_line_parameters
+
+   >>> # Define the spectrum
+   >>> np.random.seed(0)
+   >>> x = np.linspace(0., 10., 200)
+   >>> y = 3 * np.exp(-0.5 * (x - 6.3)**2 / 0.8**2)
+   >>> y += np.random.normal(0., 0.2, x.shape)
+   >>> spectrum = Spectrum1D(flux=y*u.Jy, spectral_axis=x*u.um)
+
+   >>> # Estimate parameter Gaussian1D
+   >>> g_init = estimate_line_parameters(spectrum, models.Gaussian1D()) #doctest:+SKIP
+      Model: Gaussian1D
+      Inputs: ('x',)
+      Outputs: ('y',)
+      Model set size: 1
+      Parameters:
+              amplitude            mean             stddev     
+                  Jy                um                um       
+          ----------------- ----------------- -----------------
+          3.354169257846847 6.218588636687762 1.608040201005025
+
+
+An example where the parameter estimators are defined:
```

Comment:
Maybe add a few more words here?  I.e., something about how you have to attach the estimators as `_constraints`, and that this is a behavior that may change in future versions of astropy and/or specutils (since it's a pretty ugly hack on some level).

Reply:
added